hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.dynamicconfig;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.DynamicConfigAddCacheConfigCodec;
+import com.hazelcast.config.CachePartitionLostListenerConfig;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.config.CacheSimpleConfig.ExpiryPolicyFactoryConfig;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.dynamicconfig.AddDynamicConfigOperationFactory;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.spi.OperationFactory;
+
+import java.util.List;
+
+public class AddCacheConfigMessageTask
+        extends AbstractAddConfigMessageTask<DynamicConfigAddCacheConfigCodec.RequestParameters> {
+
+    public AddCacheConfigMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected DynamicConfigAddCacheConfigCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return DynamicConfigAddCacheConfigCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return DynamicConfigAddCacheConfigCodec.encodeResponse();
+    }
+
+    @Override
+    protected OperationFactory getOperationFactory() {
+        CacheSimpleConfig config = new CacheSimpleConfig();
+        config.setAsyncBackupCount(parameters.asyncBackupCount);
```

Comment:
The package itself looks quite nice, but who really knows about it? The Spring XSD is also there and got ignored way to often. So the issue is that you need documentation what you have to change. And a `{@link}` or `{@see}` in JavaDoc is a nice way (you can click on it).

Example: I've added a new field to the `NearCacheConfig` class. So what is my to do list?
* `hazelcast-config-3.9.xsd` and `XmlConfigBuilder`
* `hazelcast-client-config-3.9.xsd` and `XmlClientConfigBuilder`
* `hazelcast-spring-3.9.xsd` (we're slowly getting used to this one)
* `NearCacheConfigReadOnly` needs an overridden setter (this is already forgotten often)
* fix `equals()`/`hashCode()` (can easily be forgotten)
* Surprise! There is the new `NearCacheConfigHolder` class, hidden in the protocol package (this is the new Spring package)

And this is just the "what is out there?" list. The next thing is:
* Which items on the list do I have to change?
* Which items am I not allowed to change?

Some config classes are IDS, some are (de)serialized in the protocol classes. So where do I have to add my new field in which way, to not break the client protocol and/or rolling upgrades. I'm already confused about this now, this won't be better the next time, I have to extend a config class.

Also: if there are classes which are not allowed to change, they should have the `@BinaryInterface` annotation.